nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/UserPrincipal.java

```diff
@@ -187,30 +188,10 @@ public boolean equals(Object o) {
 
         UserPrincipal that = (UserPrincipal) o;
 
-        // XXX: autogenerated junk, yuck!
-        if (company == null ? that.company != null : !company.equals(that.company)) {
-            return false;
-        }
-        if (firstName == null ? that.firstName != null : !firstName.equals(that.firstName)) {
-            return false;
-        }
-        if (groups == null ? that.groups != null : !groups.equals(that.groups)) {
-            return false;
-        }
-        if (lastName == null ? that.lastName != null : !lastName.equals(that.lastName)) {
-            return false;
-        }
-        if (password == null ? that.password != null : !password.equals(that.password)) {
-            return false;
-        }
-        if (roles == null ? that.roles != null : !roles.equals(that.roles)) {
-            return false;
-        }
-        if (userName == null ? that.userName != null : !userName.equals(that.userName)) {
-            return false;
-        }
-
-        return true;
+        return Objects.equals(company, that.company) && Objects.equals(firstName, that.firstName)
+                && Objects.equals(groups, that.groups) && Objects.equals(lastName, that.lastName)
+                && Objects.equals(password, that.password) && Objects.equals(roles, that.roles)
+                && Objects.equals(userName, that.userName);
     }
```

Comment:
what do you think about using org.apache.commons.lang3.builder.EqualsBuilder 
` return new EqualsBuilder().append(company, that.company).append(firstName, that.firstName)...`

same suggestion for future implementation of hashCode by using `org.apache.commons.lang3.builder.HashCodeBuilder`